plugins/packs-integrations.js

```diff
@@ -205,24 +199,21 @@ async function pluginPacksAndIntegrationsData(context, options) {
       let packDataArr = await fetchPackListItems("?limit=100", [], 0);
       console.info("completed the fetch of all the names of the pack")
       packDataArr = packDataArr.filter((pack) => {
-        return layerTypes.includes(pack.spec.layer) || (pack.spec.layer === "addon" && addOnTypes.includes(pack.spec.addonType));
+        return (((layerTypes.includes(pack.spec.layer) || (pack.spec.layer === "addon") && (addOnTypes.includes(pack.spec.addonType))) &&
+          pack.spec.registries.length && isSelectedRegistry(pack.spec.registries, mappedRepos)))
       })
       const packUrl = "v1/packs/";
       const packMDMap = new Map();
       let apiPacksData = [];
-      let promises = new Array();
-      for (let i = 0; i < packDataArr.length; i++) {
-        const packData = packDataArr[i];
-        if (packData.spec.registries.length && isSelectedRegistry(packData.spec.registries, mappedRepos)) {
-          packMDMap[packData.spec.name] = packData;
-          const cloudType = packData.spec.cloudTypes.includes("all") ? "aws" : packData.spec.cloudTypes[0];
-          promises.push(`${packUrl}${packData.spec.name}/registries/${packData.spec.registries[0].uid}?cloudType=${cloudType}&layer=${packData.spec.layer}`);
-        }
-      }
-      const results = await callRateLimitAPI(promises);
+      const urls = packDataArr.map((packData) => {
+        packMDMap[packData.spec.name] = packData;
+        const cloudType = packData.spec.cloudTypes.includes("all") ? "aws" : packData.spec.cloudTypes[0];
+        return (`${packUrl}${packData.spec.name}/registries/${packData.spec.registries[0].uid}?cloudType=${cloudType}&layer=${packData.spec.layer}`);
+      })
```

Comment:
this map should return functions that return promises directly.
we thunk the promises to avoid triggering them.